### ISSUE TITLE
[DevTools Bug]: `sayHelloToBackendManager` continuously and pointlessly polls on every page

### ISSUE THREAD

**Issue by sneakers-the-rat** [facebook/react#27992]
### Website or app

(all sites)

### Repro steps

From any page:

```js
const printMessage = (event) => {console.log(event)};
window.addEventListener('message', printMessage);
```

Since there isn't a place to actually describe the bug, i'll put it here:
`sayHelloToBackendManager` is set to poll here until `backendInitialized` is `true`: https://github.com/facebook/react/blob/b3003047101b4c7a643788a8faf576f7e370fb45/packages/react-devtools-extensions/src/contentScripts/proxy.js#L17-L23

The polling message has a source `react-devtools-content-script`: https://github.com/facebook/react/blob/b3003047101b4c7a643788a8faf576f7e370fb45/packages/react-devtools-extensions/src/contentScripts/proxy.js#L41

The switch to set `backendInitialized = true` checks for the message source `react-devtools-bridge`: https://github.com/facebook/react/blob/b3003047101b4c7a643788a8faf576f7e370fb45/packages/react-devtools-extensions/src/contentScripts/proxy.js#L65-L66

the `react-devtools-bridge` sourced message is sent from the backend by whatever invokes this: https://github.com/facebook/react/blob/b3003047101b4c7a643788a8faf576f7e370fb45/packages/react-devtools-extensions/src/contentScripts/backendManager.js#L127

but the `handleMessageFromPage` function returns if the `event.source` isn't the `window` object. Presumably the backend does not have the same window object, or else using `postMessage()` would be pointless: https://github.com/facebook/react/blob/b3003047101b4c7a643788a8faf576f7e370fb45/packages/react-devtools-extensions/src/contentScripts/proxy.js#L59

The `'react-devtools-bridge'` message never flips `backendInitialized = true`, so the polling never terminates.

This is causing the bitwarden ppl to get a decent amount of heat, and it's actually devtools causing the problem - https://github.com/bitwarden/clients/issues/7575

### How often does this bug happen?

Every time

**Comment by Nathan-Roberts123** [comment]
I don't think that the problem bitwarden/clients#7575 should be fix here in the react source code, because the problem is in the bitwarden source code.

**Comment by sneakers-the-rat** [comment]
the bug in bitwarden revealed a bug in devtools. they are both bugs.

**Comment by aescobaricc** [comment]
if you are using chrome

go to: chrome://extensions/

set option:
Site access
Allow this extension to read and change all your data on websites you visit: `On Click`

![Screenshot 2024-04-12 at 20 49 33](https://github.com/facebook/react/assets/120486257/f92e7a26-eb6e-4361-9672-785d31cf6838)

**Comment by SoCuul** [comment]
This is still happening on every site, sending constant messages on every single page.
